nx/lib/nx.ex

```diff
@@ -9391,6 +9391,74 @@ defmodule Nx do
     end
   end
 
+  def lazy_select(tensor, opts \\ []) do
+    k = opts[:k]
+
+    if rank(tensor) != 1 do
+      raise ArgumentError, "expected a vector, got a #{inspect(tensor)}"
+    end
+
+    if k >= size(tensor) do
+      raise ArgumentError, "k must be less than the size of the tensor"
+    end
+
+    len = size(tensor)
+    selection_len = :math.pow(len, 3 / 4) |> trunc()
+    opts = Keyword.put(opts, :selection_len, selection_len)
+    lazy_select_core(tensor, opts)
+  end
+
+  def lazy_select_core(tensor, opts \\ []) do
+    k = opts[:k]
+    selection_len = opts[:selection_len]
+
+    type = Nx.type(tensor)
+    max_val = Nx.Constants.max(type)
+
+    key = Nx.Random.key(System.system_time())
+    len = size(tensor)
+
+    Enum.reduce_while(1..1000, key, fn _, key ->
+      {candidates, new_key} = Nx.Random.choice(key, tensor, samples: selection_len)
+      candidates = sort(candidates)
+      x = as_type(multiply(divide(k, len), pow(len, divide(3, 4))), :s64)
```

Comment:
k/len * len ** 3/4 is the same as k * len ** (-1/4). Given that len is fixed, you could extract len ** (-1/4) from the loop and do: `x = as_type(Nx.multiply(k, len_prime), :s64)`